File: algorithms/1469/main.scala
object Solution {
    def getLonelyNodes(root: TreeNode): List[Int] = {
        (root.left, root.right) match {
            case (left: TreeNode, null) => List(left.value) ::: getLonelyNodes(left)
            case (null, right: TreeNode) => getLonelyNodes(right) ::: List(right.value)
            case (left: TreeNode, right: TreeNode) => getLonelyNodes(left) ::: getLonelyNodes(right)
            case _ => List()
        }
    }
}

File: algorithms/1469/main.go
func getLonelyNodes(root *TreeNode) []int {
    ret := []int{}
    if root.Left != nil && root.Right == nil {
        ret = append(ret, getLonelyNodes(root.Left)...)
        ret = append(ret, root.Left.Val)
    } else if root.Left == nil && root.Right != nil {
        ret = append(ret, getLonelyNodes(root.Right)...)
        ret = append(ret, root.Right.Val)
    } else if root.Left != nil && root.Right != nil {
        ret = append(ret, getLonelyNodes(root.Left)...)
        ret = append(ret, getLonelyNodes(root.Right)...)
    }
    return ret
}
